### ISSUE TITLE
`TestFunctions.testClasses` -> `TEST_CLASSES`

### ISSUE THREAD

**Issue by Juuxel** [asiekierka/MinecraftPOMF#2590]
It's a static final field.

**Comment by enbrain** [comment]
It would be great if fabric-filament can detect this, since the same mistake is happening in other places such as https://github.com/FabricMC/yarn/pull/2308#discussion_r611027554.

**Comment by Juuxel** [comment]
I added something like that to Filament locally, and it produced [a whole bunch of field naming errors](https://gist.github.com/Juuxel/0cddb23ecf02e191e60fdb50c32aee37).

I'm not 100% sure if the field naming checks should be erroring since it could make snapshot ports annoying. (imagine Mojang moving a field from static final to instance final etc.) @modmuss50 Do you have an opinion on this since you're usually the one porting?

**Comment by enbrain** [comment]
Shouldn't we consider these fields as constants? We probably need to update CONVENTIONS.md for clarification on these cases.

## non-final, but static and effectively final

```
lint: net/minecraft/util/math/Vec3f.POSITIVE_Z: non-static or non-final field starts with uppercase character 'P'
lint: net/minecraft/util/math/Vec3f.POSITIVE_X: non-static or non-final field starts with uppercase character 'P'
lint: net/minecraft/util/math/Vec3f.POSITIVE_Y: non-static or non-final field starts with uppercase character 'P'
lint: net/minecraft/util/math/Vec3f.ZERO: non-static or non-final field starts with uppercase character 'Z'
lint: net/minecraft/util/math/Vec3f.NEGATIVE_X: non-static or non-final field starts with uppercase character 'N'
lint: net/minecraft/util/math/Vec3f.NEGATIVE_Y: non-static or non-final field starts with uppercase character 'N'
lint: net/minecraft/util/math/Vec3f.NEGATIVE_Z: non-static or non-final field starts with uppercase character 'N'
```

Although they are non-final, they are static and their value is never changed after the initialization. 

## non-static, but final and effectively static

```
lint: net/minecraft/client/render/debug/ChunkLoadingDebugRenderer.LOADING_DATA_CHUNK_RANGE: non-static or non-final field starts with uppercase character 'L'
lint: net/minecraft/entity/passive/FoxEntity.CalmDownGoal.WORRIABLE_ENTITY_PREDICATE: non-static or non-final field starts with uppercase character 'W'
lint: net/minecraft/entity/mob/PhantomEntity.FindTargetGoal.PLAYERS_IN_RANGE_PREDICATE: non-static or non-final field starts with uppercase character 'P'
lint: net/minecraft/client/gui/screen/StatsScreen.ItemStatsListWidget.HEADER_ICON_SPRITE_INDICES: non-static or non-final field starts with uppercase character 'H'
lint: net/minecraft/client/gui/screen/TitleScreen.null.MULTIPLAYER_DISABLED_TEXT: non-static or non-final field starts with uppercase character 'M'
lint: net/minecraft/entity/mob/VexEntity.TrackOwnerTargetGoal.TRACK_OWNER_PREDICATE: non-static or non-final field starts with uppercase character 'T'
```

Although they are non-static, they are final and their value is same for each instance.

**Comment by Juuxel** [comment]
IMO uppercase names shouldn't be used for constant instance fields, but I agree that fields like `Vec3f`'s default instances should be considered constants. We really need a clarification in the conventions for this.

I'll introduce a severity level for the check in Filament, so that it'll still show up as a warning (since it generally is wrong to assume static fields to be constant).

**Comment by modmuss50** [comment]
I wouldnt make it fail when porting as it will just slow things down. That does make it hard to enforce, not too sure what the awswer is.

**Comment by Juuxel** [comment]
What about a workflow that only checks it on `pull_request` events? That way, direct commits aren't checked but any contributions are.

**Comment by haykam821** [comment]
This issue was fixed by #2624.
